SMLHelper/Utility/EnumCacheManager.cs

```diff
@@ -32,6 +34,42 @@ internal EnumCacheManager(string enumTypeName, int startingIndex, IEnumerable<in
             this.enumTypeName = enumTypeName;
             this.startingIndex = startingIndex;
             this.bannedIndices = bannedIndices.ToList();
+
+            // Make sure to exclude already registered TechTypes
+            if (enumTypeName.CompareTo("TechType") == 0)
+            {
+#if DEBUG
+                Logger.Log("DEBUG: Now loading known TechTypes...");
+#endif
```

Comment:
It looks really good. The only things I don't like are the `#if DEBUG` statements